### PR TITLE
chore(stark-all): upgrade version of @nbb/code-style/tslint. Refactor code to fix new TSLint violations

### PR DESCRIPTION
**Pull request by christophercr**
~~DON'T MERGE YET (https://github.com/NationalBankBelgium/code-style/pull/34/ should be merged first!)~~

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[X] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The TSLint version used in Stark is very old (5.10) 

## What is the new behavior?
- bump version of `code-style`
- bump version of `tslint` to v6.0
- fix all new linting violations as a result of the `tslint` + `code-style` upgrade

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/stark-core/tslint.json [hunk 1]
```diff
@@ -30,6 +30,63 @@
...
+    "unnecessary-else": [true, { "allow-else-if": true }], // 5.14 option added in 5.16
+    //    "no-null-undefined-union": true, 5.15
+    "no-async-without-await": true, // 5.18
+    "strict-comparisons": [true, { "allow-object-equal-comparison": true, "allow-string-order-comparison": true }], // 5.18
+    //    "no-for-in": true, // 5.19
+    "invalid-void": true, // 5.19
+    //    "strict-string-expressions": true, // 5.19
+    "no-promise-as-boolean": true // 5.19
+    // TSLint rules that were previously on tslint:recommended prior to v6.0
+    //    "arrow-parens": true,
```
Comment (christophercr) [inline packages/stark-core/tslint.json]: I've added these rules that were removed from `tslint:recommended` as from TSLint v6.

Check PR https://github.com/NationalBankBelgium/code-style/pull/34 for the explanation about this and the discussion on what we should do about it.

**Comment by SuperITMan** [comment]
@christophercr nbb/code-style@1.2.0 has been released today with the latest tslint configs.
Could you please update your PR with this new version ? 😊

**Comment by christophercr** [comment]
Oh nice! I'll update the PR then :wink:

**Comment by christophercr** [comment]
@SuperITMan PR updated... you'll see that I had to fix more linting issues as before due to the "enhancements" done in the latest releases of TSLint regarding the`typedef` rule with the `arrow-call-signature` option.

**Comment by SuperITMan** [comment]
Hey @christophercr 
Indeed, I saw your new ~140 files modified... :confused: 
Do you think we should adapt our tslint configurations to be less strict about the `arrow-call-signature` ? 
I agree that it does not make sense all the time to put it.

**Comment by christophercr** [comment]
I would suggest to keep the tslint config as we have them now because there are several cases where there is indeed a need for having the types in arrow functions. Obviously, this is not perfect as the rule requires it for all arrow functions which might not be needed all the time, but to me it doesn't harm and it is even more explicit given the current version of TypeScript we use and that we are still using TSLint.

In any case, I'm confident that this will improve later on as soon as we move to ESLint where we can have a more fine-grained configuration and to a more recent version of TypeScript 😉

**Comment by christophercr** [comment]
@SuperITMan finally... I've adapted this PR again so the `arrow-call-signature` option of `typedef` rule is disabled in all tslint configs in stark.

**Comment by SuperITMan** [comment]
@christophercr Thanks for it! 
The PR seems ok, so we can merge it 😊

**Pull request merged**